velox/docs/programming-guide/chapter01.rst

```diff
@@ -0,0 +1,376 @@
+===================================
+Chapter 1: Buffers and Flat Vectors
+===================================
+
+Velox stores and processes data one column at a time. A column of 64-bit
+integers is stored in memory using 2 contiguous buffers: a buffer of values and
+a buffer of null flags. To store 100 rows, we need to allocate values
+buffer that fits 100 64-bit integers: 100 * 8 = 800 bytes, and nulls buffer
+that fits 100 bits: 100 / 8 = 13 bytes. Velox allocates memory using
+MemoryPool.
+
+Let’s start by getting access to a MemoryPool:
+
+.. code-block:: c++
+
+    #include "velox/common/memory/Memory.h"
+
+    auto pool = memory::getDefaultMemoryPool();
+
+`pool` is a std::shared_ptr<velox::memory::MemoryPool>. We can use it to
+allocate buffers.
+
+.. code-block:: c++
+
+    #include "velox/buffer/Buffer.h"
+
+    auto values = AlignedBuffer::allocate<int64_t>(100, pool.get());
```

Comment:
A common question I have received here is how to get this type from the Velox type that we want to create the vector for. Might be worth clarifying this here.